contributors/design-proposals/taint-node-by-condition.md

```diff
@@ -0,0 +1,40 @@
+
+## Functional Detail
+Currently (1.8 and before), the conditions of nodes are updated by the kubelet and Node Controller. The kubelet updates the value to either True or False according to the node’s status. If the kubelet did not update the value, the Node Controller will set the value to Unknown after a specific grace period. 
+
+In addition to this, with taint-based-eviction, the Node Controller already taints nodes with either NotReady and Unreachable if certain conditions are met. In this proposal, the Node Controller will use additional taints on Nodes. The new taints are described below:
+
+| ConditionType      | Condition Status   |Effect        | Key      |
+| ------------------ | ------------------ | ------------ | -------- | 
+|Ready               |True                | -            | |
+|                    |False               | NoExecute    | node.alpha.kubernetes.io/notReady           |
+|                    |Unknown             | NoExecute    | node.alpha.kubernetes.io/unreachable        |
+|OutOfDisk           |True                | NoSchedule   | node.alpha.kubernetes.io/outOfDisk          |
+|                    |False               | -            | |
+|                    |Unknown             | -            | |
+|MemoryPressure      |True                | NoSchedule   | node.alpha.kubernetes.io/memoryPressure     |
+|                    |False               | -            | |
+|                    |Unknown             | -            | |
+|DiskPressure        |True                | NoSchedule   | node.alpha.kubernetes.io/diskPressure       |
+|                    |False               | -            | |
+|                    |Unknown             | -            | |
+|NetworkUnavailable  |True                | NoSchedule   | node.alpha.kubernetes.io/networkUnavailable |
+|                    |False               | -            | |
+|                    |Unknown             | -            | |
+|InodePressure       |True                | NoSchedule   | node.alpha.kubernetes.io/inodePressure      |
+|                    |False               | -            | |
+|                    |Unknown             | -            | |
+
+For example, if a CNI network is not detected on the node (e.g. a network is unavailable), the Node Controller will taint the node with `node.alpha.kubernetes.io/networkUnavailable=:NoSchedule`. This will then allow users to add a toleration to their `PodSpec`, ensuring that the pod can be scheduled to this node if necessary. If the kubelet did not update the node’s status after a grace period, the Node Controller will only taint the node with `node.alpha.kubernetes.io/unreachable`; it will not taint the node with any unknown condition.
```

Comment:
is `node.alpha.kubernetes.io/unreachable` new as well? we have to be really careful with auto-adding taints that could trigger on reachability flakes. It seems like this interacts badly with pod-eviction-timeout... if you want to wait for 5 minutes before evicting pods from an unreachable node, this would require you to add that toleration to every single pod in the system, right?